app/Http/Controllers/AttendanceController.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Attendance;
+use Illuminate\Http\Request;
+use Illuminate\Database\QueryException;
+
+class AttendanceController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('permission:read-attendance', ['only' => ['index']]);
+        $this->middleware('permission:create-attendance', ['only' => ['store']]);
+        $this->middleware('permission:read-attendance|read-attendance-own', ['only' => ['show']]);
+        $this->middleware('permission:update-attendance', ['only' => ['update']]);
+        $this->middleware('permission:delete-attendance', ['only' => ['destroy']]);
+    }
+    
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $attendance = Attendance::all();
+        return response()->json(['status' => 'success', 'attendance' => $attendance]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'attendable_type' => 'required|string',
+            'attendable_id' => 'required|numeric',
+            'gtid' => 'numeric|max:9|exists:users',
+            'source' => 'required|string',
+            'recorded_by' => 'required|numeric|exists:users,id'
+        ]);
```

Comment:
`created_at` should be accepted from the client-side, since the hardware card swipers may support an offline mode in which the data is uploaded to the API after the event is over.